core/graph/src/test/java/uk/gov/gchq/gaffer/graph/hook/NamedOperationResolverTest.java

```diff
@@ -139,13 +138,82 @@ public void shouldResolveNestedNamedOperation(@Mock final User user,
 
     @SuppressWarnings({"unchecked", "rawtypes"})
     @Test
-    public void shouldExecuteNamedOperationWithoutOverridingInput(@Mock final User user,
-                                                                  @Mock final NamedOperationCache cache,
-                                                                  @Mock final NamedOperationDetail extendedNamedOperation,
-                                                                  @Mock final GetAdjacentIds op1,
-                                                                  @Mock final GetElements op2,
-                                                                  @Mock final Iterable<? extends EntityId> input,
-                                                                  @Mock final Iterable mockIterable)
+    void shouldResolveNestedNamedOperationsWithLimit(@Mock final User user,
+                                                     @Mock final NamedOperationCache cache,
+                                                     @Mock final NamedOperationDetail namedOp1Detail,
+                                                     @Mock final NamedOperationDetail namedOp2Detail,
+                                                     @Mock final NamedOperationDetail namedOp3Detail,
+                                                     @Mock final GetAdjacentIds op1,
+                                                     @Mock final GetElements op2,
+                                                     @Mock final GetElements op3,
+                                                     @Mock final Iterable<? extends EntityId> input)
+            throws CacheOperationException {
+        // Given
+        final String namedOp1Name = "namedOp1";
+        final String namedOp2Name = "namedOp2";
+        final String namedOp3Name = "namedOp3";
+        final String namedOp4Name = "namedOp4";
+        final NamedOperationResolver resolver = new NamedOperationResolver(cache);
```

Comment:
Where is the depth value being set?
Is this only testing the default value?

Reply:
Yeah this is only testing the default depth of 3 (set in the resolver class) its not massively easy to programmatically create these nested operations so thought would be fine to just test depth of 3

Reply:
Can you change it to 2 then.
or copy and paste and change it 4.
It is to prove the you can change from the default value.

Reply:
Good point probably not a bad idea to have a test around this, though the previous implementation was missing a JSON config test too